File: C:C++/ImplementationOfStackAndQueue/StackUsingArray.cpp
#include<iostream>
using namespace std;

class Stack {
private:
	int top;
	int a[1000];

public:
	Stack(){top = -1;};
	int isEmpty();
	int isFull();
	int peek();
	void push(int d);
	int pop();
};

int Stack :: isEmpty() {
	return top == -1;
}

int Stack :: isFull() {
	return top == 999;
}

int Stack :: peek() {
	if(isEmpty())
		return -999;
	return a[top];
}

void Stack :: push(int d) {
	if(isFull()) {
		cout << "Stack is Full\n";
	}else{
		top ++;
		a[top] = d;
	}
}

int Stack :: pop() {
	if(isEmpty()) {
		cout << "Stack is Empty\n";
		return -999;
	}else{
		int temp = a[top];
		top --;
		return temp;
	}
}


int main() {

	Stack s;

	cout << s.pop() << endl;

}

File: Java/ImplementationOfStackAndQueue/StackUsingArray.java
class StackUsingArray {

	private int top;
	private int capacity;
	private int[] a;

	public StackUsingArray(int capacity) {
		this.capacity = capacity;
		this.top = -1;
		a = new int[this.capacity];
	}

	public StackUsingArray() {
		this(5);
	}

	public boolean isFull() {
		return top == capacity - 1;
	}

	public boolean isEmpty() {
		return top == -1;
	}

	public void push(int d) {
		if(isFull())
			throw new IllegalArgumentException("Stack Overflow");
		a[++top] = d;
	}

	public int pop() {
		if(isEmpty())
			throw new IndexOutOfBoundsException("Stack Underflow");
		int temp = a[top];
		--top;
		return temp;
	}

	public int peek() {
		if(isEmpty())
			throw new IndexOutOfBoundsException("Stack is Empty");
		return a[top];
	}

}